gen/_template/security.tmpl

```diff
@@ -74,6 +86,13 @@ func (s *Server) security{{ $s.Type.Name }}(ctx context.Context, operationName s
 			return ctx, false, nil
 		}
 		t.Token = token
+	{{- else if $s.Format.IsOAuth2Security }}
+		token, ok := findAuthorization(req.Header, "Bearer")
+		if !ok {
+			return ctx, false, nil
+		}
+		t.Token = token
+		t.Scopes = oauth2Scopes[operationName]
```

Comment:
@tdakkota is an arbitrary `operationName` (not present in the spec and code) possible here?
If so, we need to handle this case by making sure that the oauth scopes for `operationName` exist in the map.

Reply:
I was thinking of this case but then realized that with go and how it handles maps its not a problem if it won't exist. If map key doesn't exist it returns just a default version of the value type. So I thought it's not really needed to add extra logic.

Example:
https://go.dev/play/p/autilf-GwdK

Reply:
Meaning if operationName doesn't exist it Scopes will be set to an empty list.

Reply:
Sure, but I'm more concerned about is it ok to pass empty oauth scopes to security handler?
Does oauth2 spec allow that?

Reply:
Yeah was wondering the same thing when implementing. According to the spec here: https://swagger.io/docs/specification/authentication/oauth2/.

```
No Scopes
Scopes are optional, and your API may not use any. In this case, specify an empty object {} in the scopes definition, and an empty list of scopes [] ...
```

Meaning its up to the spec security implementator to decide if he wants to use scopes or not 🤔

Reply:
Seems fine - https://swagger.io/docs/specification/authentication/oauth2/